src/impl/vamp/planning/rrt_star.hh

```diff
@@ -0,0 +1,285 @@
+
+                    // rewire with our newly added node to if we can use it to get a shorter path to any of
+                    // it's neighbors
+                    idx = 0;
+                    bool check_new_best = false;
+                    for (auto &[node, distance] : near)
+                    {
+                        if (collision_free[idx] and (min_cost + distance < cost[node.index]))
+                        {
+                            remove_edge(parent[node.index], node.index);
+                            add_edge(free_index, node.index);
+
+                            // propgate saved cost to children
+                            cost[node.index] = min_cost + distance;
+                            update_children(node.index);
+                            check_new_best = true;
+                        }
+                        idx++;
+                    }
+
+                    free_index++;
+
+                    bool goal_reached = false;
+                    // check if we can reach a goal
+                    for (const auto &goal : goals)
+                    {
+                        if (validate_motion<Robot, rake, resolution>(new_configuration, goal, environment))
+                        {
+                            if (not settings.force_max_iters)
+                            {  // found a solution, exit the algorithm
+                                auto current = free_index - 1;
+                                result.path.emplace_back(goal);
+                                result.path.emplace_back(new_configuration);
+                                result.cost = cost[current] + goal.distance(buffer_index(current));
+                                while (parent[current] != current)  // build path
+                                {
+                                    auto parent_node = parent[current];
+                                    result.path.emplace_back(buffer_index(parent_node));
+                                    current = parent_node;
+                                }
+                                std::reverse(result.path.begin(), result.path.end());
+                                goal_reached = true;
+                                break;
+                            }
+                            else
+                            {
+                                goal_motions.emplace_back(free_index - 1, goal);
+                                check_new_best = true;
+                            }
+                        }
+                    }
+                    if (not settings.force_max_iters and goal_reached)
+                    {
+                        break;
+                    }
+                    if (settings.force_max_iters and check_new_best)
+                    {
+                        // loop through goal nodes to find the best solution
+                        for (auto &[end_node_idx, goal] : goal_motions)
```

Comment:
const

Reply:
Why are you tracking multiple goal motions? You should just be able to keep the best one.